test/norm_num.lean

```diff
@@ -4,216 +4,459 @@ Released under Apache 2.0 license as described in the file LICENSE.
+end OrderedCharZeroRing
+
+section LinearOrderedRing
+
+variable [LinearOrderedRing α]
+
+example : (1 : α) ≤ 1 := by norm_num1
+example : (-1 : α) ≤ -1 := by norm_num1
+example : (-1 : α) ≤ 1 := by norm_num1
+
+example : (1 : α) < 2 := by norm_num1
+example : (-1 : α) < 2 := by norm_num1
+example : (-1 : α) < 1 := by norm_num1
+
+end LinearOrderedRing
+
+section Rat
+
+variable [LinearOrderedField α] [Nontrivial α]
+
+-- Normalize to True
+example : (1 : ℚ) ≤ 1 := by norm_num1
+example : (-1 : ℚ) ≤ -1 := by norm_num1
+example : (-1 : ℚ) ≤ 1 := by norm_num1
+example : (-1/2 : ℚ) ≤ 1 := by norm_num1
+example : (1/2 : ℚ) ≤ 5/4 := by norm_num1
+example : (1 : α) ≤ 1 := by norm_num1
+example : (-1 : α) ≤ -1 := by norm_num1
+example : (-1 : α) ≤ 1 := by norm_num1
+example : (-1/2 : α) ≤ 1 := by norm_num1
+example : (1/2 : α) ≤ 5/4 := by norm_num1
+
+example : (1 : ℤ) < 2 := by norm_num1
+example : (-2 : ℤ) < 1 := by norm_num1
+example : (-1 : ℤ) < 1 := by norm_num1
+example : (-1/2 : ℚ) < 1 := by norm_num1
+example : (1/2 : ℚ) < 5/4 := by norm_num1
+example : (1 : α) < 2 := by norm_num1
+example : (-1 : α) < 2 := by norm_num1
+example : (-1 : α) < 1 := by norm_num1
+example : (-1/2 : α) < 1 := by norm_num1
+example : (1/2 : α) < 5/4 := by norm_num1
+
+-- Normalize to False
+example : ¬((2 : ℚ) ≤ 1) := by norm_num1
+example : ¬((2 : ℚ) ≤ -1) := by norm_num1
+example : ¬((1/2 : ℚ) ≤ -1) := by norm_num1
+example : ¬((2 : α) ≤ 1) := by norm_num1
+example : ¬((2 : α) ≤ -1) := by norm_num1
+example : ¬((1/2 : α) ≤ -1) := by norm_num1
 
--- example : (1:ℂ) ≠ 2 := by norm_num
--- example : (1:ℂ) / 3 ≠ 2 / 7 := by norm_num
+example : ¬((2 : ℚ) < 1) := by norm_num1
+example : ¬((2 : ℚ) < -1) := by norm_num1
+example : ¬((1/2 : ℚ) < -1) := by norm_num1
+example : ¬((2 : α) < 1) := by norm_num1
+example : ¬((2 : α) < -1) := by norm_num1
+example : ¬((1/2 : α) < -1) := by norm_num1
 
--- example : (1:ℝ) ≠ 2 := by norm_num
+end Rat
 
--- example {α} [Semiring α] : (1:α) ≠ 2 := by norm_num -- [fixme]
--- example {α} [Ring α] : (-1:α) ≠ 2 := by norm_num -- [fixme]
--- example {α} [DivisionRing α] [CharZero α] : (-1:α) ≠ 2 := by norm_num
--- example {α} [DivisionRing α] [CharZero α] : (1:α) / 3 ≠ 2 / 7 := by norm_num
--- example {α} [DivisionRing α] [CharZero α] : (1:α) / 3 ≠ 0 := by norm_num
+end Order
 
-example : (5 / 2:ℕ) = 2 := by norm_num
-example : (5 / -2:ℤ) < -1 := by norm_num
-example : (0 + 1) / 2 < 0 + 1 := by norm_num
-example : Nat.succ (Nat.succ (2 ^ 3)) = 10 := by norm_num
--- example : 10 = (-1 : ℤ) % 11 := by norm_num -- [fixme] ⊢ False ???
-example : (12321 - 2 : ℤ) = 12319 := by norm_num
--- example : (63:ℚ) ≥ 5 := by norm_num
+example : (1:ℂ) ≠ 2 := by norm_num1
+example : (1:ℂ) / 3 ≠ 2 / 7 := by norm_num1
+
+example : (1:ℝ) ≠ 2 := by norm_num1
+
+-- example : (5 / 2:ℕ) = 2 := by norm_num1
+-- example : (5 / -2:ℤ) < -1 := by norm_num1
+-- example : (0 + 1) / 2 < 0 + 1 := by norm_num1
+-- example : Nat.succ (Nat.succ (2 ^ 3)) = 10 := by norm_num1
```

Comment:
These were commented out as well, why?

Reply:
this involves nat division, int division, and nat.succ, none of which are implemented